docs/configuration/options.md

```diff
@@ -57,20 +61,22 @@ See the [providers documentation](/configuration/providers/oauth) for a list of
 
 A random string is used to hash tokens, sign/encrypt cookies and generate cryptographic keys.
 
-If not specified in development, it uses a hash for all configuration options, including OAuth Client ID / Secrets for entropy. Although if the user does not use such a provider, the configuration might be guessed.
+If you set [`NEXTAUTH_SECRET`](#nextauth_secret) as an environment variable, you don't have to define this option.
+
+If no value specified specified in development (and there is no `NEXTAUTH_SECRET` variable either), it uses a hash for all configuration options, including OAuth Client ID / Secrets for entropy.
 
-You can quickly create a valid secret on the command line via this `openssl` command.
+:::warning
+Not providing any `secret` or `NEXTAUTH_SECRET` will throw [an error](/errors#no_secret) in production.
+:::
+
+You can quickly create a good value on the command line via this `openssl` command.
 
 ```bash
 $ openssl rand -base64 32
 ```
 
-:::warning
-The default behaviour is volatile, and it is strongly recommended you explicitly specify a value. If `secret` is omitted in production, an error is thrown.
-:::
-
 :::tip
-If you rely on the default secret generation in development, you might notice JWT decryption errors, since the secret changes whenever you change your configuration. Defining a secret will make this problem go away.
+If you rely on the default secret generation in development, you might notice JWT decryption errors, since the secret changes whenever you change your configuration. Defining a secret will make this problem go away. We will likely make this option mandatory even in development in the future.
```

Comment:
```suggestion
If you rely on the default secret generation in development, you might notice JWT decryption errors, since the secret changes whenever you change your configuration. Defining a secret will make this problem go away. We will likely make this option mandatory, even in development, in the future.
```